asset/asset.go

```diff
@@ -2001,7 +2013,14 @@ func (a *Asset) Decode(r io.Reader) error {
 		return err
 	}
 
-	return TlvStrictDecode(stream, r, KnownAssetLeafTypes)
+	unknownOddTypes, err := TlvStrictDecode(stream, r, KnownAssetLeafTypes)
+	if err != nil {
+		return err
+	}
+
+	a.unknownOddTypes = unknownOddTypes
```

Comment:
Dig the implementation! 

One gap (still getting through the PR) that may pop up is when we encode a protocol object within the database as a fully featured row vs encoding an encoded blob.

Reply:
Isn't that what we do for most fields?

Reply:
We wouldn't actually _store_ the information in the unknown types. Just keep it in memory so we can validate correctly (e.g. arrive at the correct leaf hash for assets or the correct meta hash for meta reveals).
Only when you upgrade to the version that implements the new value, would we store it, with the corresponding database migration. But then it would be a known type and the value wouldn't end up in `unknownOddTypes` anymore.